app/controllers/application_controller.rb

```diff
@@ -31,4 +32,18 @@ def collect_metrics
     duration = Time.zone.now - start
     Rails.logger.info "#{controller_name}##{action_name}: #{duration}s"
   end
+
+  private
+
+  def set_locale
+    I18n.locale = locale
+  end
+
+  def locale
```

Comment:
You might wonder why the `LocaleExtractor` call isn't made inside the `set_locale` method directly. This method is extracted so we could extend it by checking if there's a current user and using `current_user.language`.

I actually tried it.

```ruby
  def locale
    current_user&.language || LocaleExtractor.new(accept_lang_header).extract
  end
```

But for some reason, it doesn't work when trying to log in with an unconfirmed account. [The line where the locale is set](https://github.com/pieforproviders/pieforproviders/pull/219/files#diff-55c5b7aecfb519d0e4880eaf2788eb6eR39) doesn't get executed on that request. It's as if the response ends with a 401 when calling `current_user`. If you know why this is happening, let me know.

Reply:
I wonder if, because you can't log in with an unconfirmed account, that "action" is never considered to be completed, so the hook isn't called. 🤔

Reply:
Looks like it. It just halts the chain after figuring the user has an unconfirmed account.